include/SFML/Graphics/StencilMode.hpp

```diff
@@ -0,0 +1,251 @@
+    GreaterEqual, //!< The stencil test passes if the new value is greater than or equal to the value in the stencil buffer
+    Equal,        //!< The stencil test passes if the new value is strictly equal to the value in the stencil buffer
+    NotEqual,     //!< The stencil test passes if the new value is strictly inequal to the value in the stencil buffer
+    Always        //!< The stencil test always passes
+};
+
+////////////////////////////////////////////////////////
+/// \brief Enumeration of the stencil buffer update operations
+///
+/// The update operations are mapped directly to their OpenGL equivalents,
+/// specified by glStencilOp().
+////////////////////////////////////////////////////////
+enum class StencilUpdateOperation
+{
+    Keep,      //!< If the stencil test passes, the value in the stencil buffer is not modified
+    Zero,      //!< If the stencil test passes, the value in the stencil buffer is set to zero
+    Replace,   //!< If the stencil test passes, the value in the stencil buffer is set to the new value
+    Increment, //!< If the stencil test passes, the value in the stencil buffer is incremented and if required clamped
+    Decrement, //!< If the stencil test passes, the value in the stencil buffer is decremented and if required clamped
+    Invert,    //!< If the stencil test passes, the value in the stencil buffer is bitwise inverted
+};
+
+////////////////////////////////////////////////////////////
+/// \brief Stencil modes for drawing
+///
+////////////////////////////////////////////////////////////
+struct SFML_GRAPHICS_API StencilMode
+{
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    /// Constructs a stencil mode that disables stencil testing
+    ///
+    ////////////////////////////////////////////////////////////
+    StencilMode();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Construct a stencil mode given the function, operation and reference value.
+    ///
+    /// \param theStencilComparison      Specifies the kind of test to perform against the value in the stencil buffer
+    /// \param theStencilUpdateOperation Specifies how the stencil buffer is updated if the test passes
+    /// \param theStencilReference       Specifies the reference value to perform the stencil test with and to write into stencil buffer if the operation is Replace and the test passes
+    /// \param theStencilMask            Specifies the mask value to apply to both the reference value and the value in the stencil buffer when comparing and updating
+    /// \param theStencilOnly            True to update only the stencil buffer, false to update both stencil and color buffers
+    ///
+    ////////////////////////////////////////////////////////////
+    StencilMode(StencilComparison      theStencilComparison,
```

Comment:
This was done analogous to the `RenderStates` constructor. If we want to clean this up at some later point we should do it all at once instead of one at a time.

Reply:
I don't think repeating the mistakes of `RenderStates` is the right call here. We're already so close to have an aggregate type. We just have to delete this single constructor. `RenderStates` has 6 constructors of which we'd have to remove all 6 to make it an aggregate.

Reply:
https://github.com/SFML/SFML/compare/master...aggregate_render_states

Turns out it's not so hard to remove all those `RenderStates` constructors. At least internally we had little use for many of those constructors.